### PR TITLE
Add Building a RESTful API with Node.js

### PR DESCRIPTION
**Pull request by SidhaantThakker**
## What does this PR do?
Add resource(s)

### Description
Adds Building a RESTful API with Node.js, a playlist on YouTube by Academind.

### Why is this valuable (or not)?
This course covers all the elements of building a Node.js RESTful API, from basic route creating to advanced concepts like route protection and user signup, which is essential for anyone looking to learn the popular MERN stack.

### How do we know it's really free?
It's uploaded by the creator, Academind, on YouTube for free for everyone and does not require payment/donation to view.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a course.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by SidhaantThakker** [comment]
If I discover any more good free resources in the future can I open more pull requests? I don't want to seem like I'm spamming :)

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -448,6 +448,7 @@
 #### NodeJS
 
 * [Beginner's Series to: Node.js](https://youtube.com/playlist?list=PLlrxD0HtieHje-_287YJKhY8tDeSItwtg) - Microsoft Developer (YouTube)
+* [Building a RESTful API with Node.js](https://www.youtube.com/playlist?list=PL55RiY5tL51q4D-B63KBnygU6opNPFk_q) - Academind (Youtube)
```
Comment (davorpa) [inline courses/free-courses-en.md:451]: Add @academind complementary material

```markdown
* [Building a RESTful API with Node.js](https://www.youtube.com/playlist?list=PL55RiY5tL51q4D-B63KBnygU6opNPFk_q) - Academind (video and [:package: ])(Youtube)
```

**Review by davorpa — CHANGES_REQUESTED** [review]
> If I discover any more good free resources in the future can I open more pull requests? I don't want to seem like I'm spamming :)

Yes, of course, but with calm and always following the CONTRIBUTING guidelines :wink:

Please, apply suggestions

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

File: courses/free-courses-en.md (hunk 1 above)
Comment (SidhaantThakker) [reply courses/free-courses-en.md:451]: I'll commit the change now?

**Pull request merged**